cli/src/base.ts

```diff
@@ -1,54 +1,74 @@
-import { Command } from '@oclif/core';
+import { Command, Flags } from '@oclif/core';
 import { XataApiClient } from '@xata.io/client';
 import ansiRegex from 'ansi-regex';
 import chalk from 'chalk';
+import { cosmiconfigSync } from 'cosmiconfig';
 import dotenv from 'dotenv';
+import { readFile, writeFile } from 'fs/promises';
 import fetch from 'node-fetch';
+import path from 'path';
+import prompts from 'prompts';
+import slugify from 'slugify';
 import table from 'text-table';
-import { z } from 'zod';
+import { z, ZodError } from 'zod';
 import { readAPIKey } from './key.js';
-import { cosmiconfigSync } from 'cosmiconfig';
 
-export const projectConfig = z.object({
-  codegen: z.optional(
-    z.object({
-      output: z.string()
-    })
-  )
+export const projectConfigSchema = z.object({
+  databaseURL: z.string(),
+  codegen: z.object({
+    output: z.string(),
+    declarations: z.boolean()
+  })
 });
 
-export type ProjectConfig = z.infer<typeof projectConfig>;
+const partialProjectConfig = projectConfigSchema.deepPartial();
```

Comment:
It just changes where the deepPartial() is, right? The reason to put it there is because I want the schema without the optionals for the `config set` command. With the non-optionals schema I can easily traverse the schema to get the type of a specific config setting. With the partial schema it was more complicated and verbose. So basically I have both versions to different use cases.